### ISSUE TITLE
Compatibility of popup with the Picom compositor

### ISSUE THREAD

**Issue by Flashwalker** [M7S/dockbarx#173]
Hi

Is there any chance to make some how a compatibility of transparency around the popup with the **Picom** compositor?

What I mean is that the shadow and blur applied by **Picom** are not accurately applied to the popup border shape, but to the rectangle around it.

![image](https://user-images.githubusercontent.com/2188215/216286963-8b459790-f36c-4121-a993-36b33147def6.png)

link for ref: https://github.com/yshui/picom/discussions/1016

**Comment by xuzhen** [comment]
Fixed in https://github.com/xuzhen/dockbarx/commit/e05d687c57a7fbffca09e9af4148d19348e39ea8
Enable `Use X11 shape extension for the popup` option in the `Window List` tab

**Comment by Flashwalker** [comment]
> Fixed in [xuzhen@e05d687](https://github.com/xuzhen/dockbarx/commit/e05d687c57a7fbffca09e9af4148d19348e39ea8) Enable `Use X11 shape extension for the popup` option in the `Window List` tab

Is a PPA update pending?
I tried to install via pip:
```bash
~/dockbarx.git $ sudo python3 -m pip install .
Processing /home/user/dockbarx.git
  Preparing metadata (setup.py) ... done
Building wheels for collected packages: Dockbarx
  Building wheel for Dockbarx (setup.py) ... done
  Created wheel for Dockbarx: filename=Dockbarx-1.0b2-py3-none-any.whl size=619446 sha256=0ff3b866fe9a9789ed77f39a3f3f39041d1430cb20b4a30071d43860cd9bd066
  Stored in directory: /tmp/pip-ephem-wheel-cache-fddhvjzq/wheels/7a/6e/12/49148bf968bfeb02a79718410ed1d2e553c05bf5b1b7ae934c
Successfully built Dockbarx
Installing collected packages: Dockbarx
  Attempting uninstall: Dockbarx
    Found existing installation: Dockbarx 1.0b2
    Uninstalling Dockbarx-1.0b2:
      Successfully uninstalled Dockbarx-1.0b2
Successfully installed Dockbarx-1.0b2
```
but it fails :
```
$ dockx
GLib-GIO-ERROR **: 23:39:58.258: Settings schema 'org.dockbarx.dockbarx' is not installed
Trace/breakpoint trap (core dumped)
```

**Comment by Flashwalker** [comment]
> Fixed in [xuzhen@e05d687](https://github.com/xuzhen/dockbarx/commit/e05d687c57a7fbffca09e9af4148d19348e39ea8) Enable `Use X11 shape extension for the popup` option in the `Window List` tab

I found that to make it works you should run this after installation:
```bash
sudo glib-compile-schemas /usr/share/glib-2.0/schemas/
```

and set this in your `~/.config/picom.conf`:
```bash
shadow-ignore-shaped = false;
```

Now it works, thanks

**Comment by Flashwalker** [comment]
The only thing buggy is that  this xshape thing start to works only after i moving the mouse cursor into the windows list.  
It's still bounded with rectangle until i do it

**Comment by xuzhen** [comment]
I think it's picom's fault. compton can draw the shadow correctly even without the shape extension.

**Comment by Flashwalker** [comment]
> I think it's picom's fault. compton can draw the shadow correctly even without the shape extension.

I meant this behavior (screencast with compton):

https://user-images.githubusercontent.com/2188215/227863085-02f63caf-751b-4628-94ac-22a13f2e95f2.mp4

xshape applied only after second mouse hover event or on the second appearance of the popup
(for newly oppened apps)

**Comment by xuzhen** [comment]
try https://github.com/xuzhen/dockbarx/commit/f71b55498c4f11b6cfa9f46e5aebdf4d0c9831bc

**Comment by Flashwalker** [comment]
> try [xuzhen@f71b554](https://github.com/xuzhen/dockbarx/commit/f71b55498c4f11b6cfa9f46e5aebdf4d0c9831bc)

Cool, now it works fine right away, but
now another problem appeared with the context menu (right-click). Here it is:

https://user-images.githubusercontent.com/2188215/228524777-6c6f2597-4a44-4887-95b9-8b994f6c504f.mp4

**Comment by xuzhen** [comment]
Should be fixed in https://github.com/xuzhen/dockbarx/commit/71a38e8956ab3aec38733ab01795e6e21735488c

**Comment by Flashwalker** [comment]
> Should be fixed in [xuzhen@71a38e8](https://github.com/xuzhen/dockbarx/commit/71a38e8956ab3aec38733ab01795e6e21735488c)

Yes, it's ok now, thanks
